src/doc/trpl/dining-philosophers.md

```diff
@@ -0,0 +1,942 @@
+```
+
+The `use` keyword lets us bring names into scope. In this case, we are going to
+use two things: the `mspc` module from `std::sync`, and the `Sender` type from
+said module. As you may have guessed, `mspc` stands for something:
+multi-producer, single consumer. We’re going to be using Rust’s ‘channel’ types
+to handle the concurrency here, and they’re in this part of the standard
+library.
+
+```rust,ignore
+struct Philosopher {
+    name: String,
+    done: Sender<bool>
+}   
+```
+
+When our philosopher is done eating, they’re going to send a boolean value
+down a channel. The sending end of a channel has the `Sender` type, which
+we `use`d in the last line.
+
+```rust,ignore
+impl Philosopher { 
+    fn new(name: &str, done: Sender<bool>) -> Philosopher {
+        Philosopher {
+            name: name.to_string(),
+            done: done,
+        }
+    }
+```
+
+Now that we have a second field for our struct, we need to define it in
+`new()`. In this case, we’re going to take it as an argument. We’ll pass this
+in in `main()`, below.
+    
+```rust,ignore
+    fn done(&self) {
+        println!("{} is done eating.", self.name);
+
+        self.done.send(true).ok().expect("Couldn't finish eating");
+    }
+}
+```
+
+We’ve added a new line to `done()`. We use the `send()` method on `Sender`
+to send `true` down the channel. `send()` returns a `Result`, letting us
+know if we’ve had some sort of error while trying to send the value. In
+this case, we want to just crash and throw an error if it doesn’t work,
+so we use the `ok().expect()` pattern we used in the Guessing Game.
+
+```rust,ignore
+fn main() {
+    let (done_tx, done_rx) = mpsc::channel();
+```
+
+We need to actually create a channel, and the `mpsc::channel()` function does
+that. It returns both ends of the channel, the sending and receiving end. We
+can use a single `let` to bind both ends at the same time, setting both
+`done_tx` and `done_rx`.
+
+```rust,ignore
+    let philosophers = vec![
+        Philosopher::new("Baruch Spinoza", done_tx.clone()),
+        Philosopher::new("Gilles Deleuze", done_tx.clone()),
+        Philosopher::new("Karl Marx", done_tx.clone()),
+        Philosopher::new("Friedrich Nietzsche", done_tx.clone()),
+        Philosopher::new("Michel Foucault", done_tx.clone()),
+    ];
+```
+
+Now that we’ve gotten a channel, we want each philosopher to have a copy of the
+sending end. The `clone()` method on `Sender` lets us do that. Now we have a
+channel with the receiving end in `main()`, and the sending end in each one of
+our `Philosopher`s.
+
+```rust,ignore
+    for _ in 0..5 {
+        done_rx.recv().unwrap();
+    }
+}
+```
+
+The `recv()` method is on the receiving end of the channel, which has the type
+`Receiver`. Since we have five philosophers, we loop five times, and ask for
+five values on the channel. This all matches up, and our program runs. You should
+get the same output as before: all five philosophers reporting in.
```

Comment:
You already noted I have my doubts about using a channel here, but in this context I find it particularly dubious.
The value send is always `true`, so there is no information transmitted here, you know it's going to be true, zero entropy etc., etc. So you could as well be sending `()`, which conveys that there is no information. Why would you do that though? (Something this sections doesn't answer) Well, usually for the side-effect. With threads you can use this to create a barrier, making sure all threads have finished before continuing. Since this is before introducing threads though, I'd argue that it is at the very least too early in the guide.